### PR TITLE
Convert fields store to setup function

### PR DESCRIPTION
**Pull request by nickrum**

### CODE REVIEW

**Review by rijkvanzanten — APPROVED** [review]

**Comment by rijkvanzanten** [comment]
@nickrum I'll let you handle the tests 🚀

File: app/src/stores/fields.ts [hunk 1]
```diff
@@ -65,308 +66,338 @@ const fakeFilesField: Field = {
...
-				if (groupFields.length) {
-					nonGroupFields.splice(index + 1, 0, ...orderBy(groupFields, 'meta.sort'));
-				}
-			}
+	/**
+	 * Retrieve field info for a (deeply) nested field
+	 * Recursively searches through the relationships to find the field info that matches the
+	 * dot notation.
+	 */
+	function getRelationalField(collection: string, fields: string) {
```
Comment (paescuj) [inline app/src/stores/fields.ts:379]: Does this function not need to be exposed?

Comment (nickrum) [reply app/src/stores/fields.ts:379]: It isn't used anywhere except inside this store as far as I can see.

**Comment by nickrum** [comment]
> I'll let you handle the tests 🚀

Thanks 🥲 Now that functions/actions in the store aren't called through the `this` context of the store, you can no longer spy on them being called inside the store, so I had to workaround that.

**Review by rijkvanzanten — APPROVED** [review]

**Review by paescuj — APPROVED** [review]

**Pull request merged**